__snapshots__/packages/java-edition/test-out/mcfunction/parser/argument/minecraftNbtTag.spec.js

```diff
@@ -24,152 +33,197 @@ exports['mcfunction argument minecraft:nbt_tag Parse ""foo"" 1'] = {
 
 exports['mcfunction argument minecraft:nbt_tag Parse "0" 1'] = {
   "node": {
-    "type": "nbt:int",
+    "type": "mcfunction:nbt",
     "range": {
       "start": 0,
       "end": 1
     },
-    "value": 0
+    "children": [
+      {
+        "type": "nbt:int",
+        "range": {
+          "start": 0,
+          "end": 1
+        },
+        "value": 0
+      }
+    ]
   },
   "errors": []
 }
 
 exports['mcfunction argument minecraft:nbt_tag Parse "0.0" 1'] = {
   "node": {
-    "type": "nbt:double",
+    "type": "mcfunction:nbt",
     "range": {
       "start": 0,
       "end": 3
     },
-    "value": 0
+    "children": [
+      {
+        "type": "nbt:double",
+        "range": {
+          "start": 0,
+          "end": 3
+        },
+        "value": 0
+      }
+    ]
   },
   "errors": []
 }
 
 exports['mcfunction argument minecraft:nbt_tag Parse "0b" 1'] = {
   "node": {
-    "type": "nbt:byte",
+    "type": "mcfunction:nbt",
     "range": {
       "start": 0,
       "end": 2
     },
-    "value": 0
+    "children": [
+      {
+        "type": "nbt:byte",
+        "range": {
+          "start": 0,
+          "end": 2
+        },
+        "value": 0
+      }
+    ]
   },
   "errors": []
 }
 
 exports['mcfunction argument minecraft:nbt_tag Parse "0l" 1'] = {
   "node": {
-    "type": "nbt:long",
+    "type": "mcfunction:nbt",
     "range": {
       "start": 0,
       "end": 2
     },
-    "value": "0"
+    "children": [
+      {
+        "type": "nbt:long",
+        "range": {
+          "start": 0,
+          "end": 2
+        },
+        "value": "0"
```

Comment:
Why is the value for a long stored as a string compared to doubles, bytes, and ints?

Reply:
Additionally it might be a good idea to have tests for 1.0f, 1.0d and maybe a negative number but it's not too important

Reply:
JavaScript uses 64 bit IEEE floating point numbers which does not have enough precision to store a 64 bit long integer, so an arbitrary-precision integer type `BitInt` is used instead, which gets serialized to strings in our snapshot implementation.

Reply:
Okay got it, just wanted to make sure it wasn't a mistake :+1: